src/base/net/downloadhandler.cpp

```diff
@@ -174,29 +179,30 @@ void Net::DownloadHandler::handleRedirection(QUrl newUrl)
             emit downloadFailed(m_downloadRequest.url(), tr("Unexpected redirect to magnet URI."));
 
         this->deleteLater();
+        return;
     }
-    else {
-        DownloadHandler *redirected = m_manager->download(DownloadRequest(m_downloadRequest).url(newUrlString));
-        connect(redirected, &DownloadHandler::destroyed, this, &DownloadHandler::deleteLater);
-        connect(redirected, &DownloadHandler::downloadFailed, this, [this](const QString &, const QString &reason)
-        {
-            emit downloadFailed(url(), reason);
-        });
-        connect(redirected, &DownloadHandler::redirectedToMagnet, this, [this](const QString &, const QString &magnetUri)
-        {
-            emit redirectedToMagnet(url(), magnetUri);
-        });
-        connect(redirected, static_cast<void (DownloadHandler::*)(const QString &, const QString &)>(&DownloadHandler::downloadFinished)
-                , this, [this](const QString &, const QString &fileName)
-        {
-            emit downloadFinished(url(), fileName);
-        });
-        connect(redirected, static_cast<void (DownloadHandler::*)(const QString &, const QByteArray &)>(&DownloadHandler::downloadFinished)
-                , this, [this](const QString &, const QByteArray &data)
-        {
-            emit downloadFinished(url(), data);
-        });
-    }
+
+    const DownloadHandler *redirected = m_manager->download(
+        DownloadRequest(m_downloadRequest).url(newUrlString).redirectionCounter(redirectCount + 1));
```

Comment:
Sorry, I have had too little free time in the last few days, so my previous comments were a bit spontaneous and not very correct. This place rubbed me the wrong way, but I'm not fully aware of what is going on here. I understand that now.
DownloadRequest encapsulates "download request" properties, it just packs parameters of `DownloadManager::download()` method, i.e. it is part of DownloadManager public interface. But "redirection counter" is part of DownloadHandler internals and it shouldn't be the properly of DownloadRequest (since "download request" hasn't such a property in its essence).
To fix this issue you just need to have it as private field in DownloadHandler class and increment it here directly.

Reply:
Done, updated.